CMakeLists.txt

```diff
@@ -192,21 +197,48 @@ else()
 
 endif()  # !MSVC
 
-add_library(hwy STATIC ${HWY_SOURCES})
+# By default prefer STATIC build (legacy behavior)
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
+# only expose shared option to advanced users:
+mark_as_advanced(BUILD_SHARED_LIBS)
+# Define visibility settings globally:
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+
+# This preprocessor define will drive the build, also used in the *.pc files:
+if(BUILD_SHARED_LIBS)
+  set(HWY_DLLEXPORT_STRING "HWY_SHARED_DEFINE")
```

Comment:
We might consider renaming? Generic terms such as string or manager don't add much value. Maybe something like HWY_WHICH_DLLEXPORT or DLLEXPORT_TO_DEFINE?

Reply:
Sure, why not. I picked `DLLEXPORT_TO_DEFINE`.